### PR TITLE
URL functionality

### PR DESCRIPTION
**Pull request by doc-han**
Enabling the module to take YouTube Video URL instead of ID.

### CODE REVIEW

**Review by displaynone — CHANGES_REQUESTED** [review]
You shouldn't change that function, instead you should add a new function `getVideoInfoByURL( url )` that, using oEmbed, will get the id and then the data from `getVideoInfo`.

Keep in mind that there are several types of YouTube URLs and your code doesn't work for each of them.

**Comment by Doc-Han** [comment]
kk I have a Regex that covers most of th URL's.

**Comment by Doc-Han** [comment]
I think you can test for various URL's. Or should I add test script?

**Comment by Doc-Han** [comment]
List of example youtube URL's [here](https://gist.github.com/Glurt/ea11b690ba4b1278e049)

**Comment by displaynone** [comment]
I think you are not using `.editorconfig` or `.eslintrc`

**Comment by Doc-Han** [comment]
Sorry I'm a beginner who just moved to Intermediate

**Comment by Doc-Han** [comment]
I think it's linted now

**Comment by displaynone** [comment]
Don't be sorry for that, nobody's born knowing ;)
Glad to help you and willing to merge your PR :)

**Comment by Doc-Han** [comment]
eslint said I don't need that escape but I do for the regex to work

**Comment by Doc-Han** [comment]
Any additional thing I'm to  do?

**Review by displaynone — APPROVED** [review]
@Doc-Han please, review the changes I have implemented

Thanks for your contribution :)

Hope to "see" you again :)

File: index.js [hunk 1]
```diff
@@ -51,6 +51,15 @@ class YoutubeDownloader {
     } );
   }
 
+  getVideoInfoByURL( link ){
+    link = decodeURIComponent(link)+ " ";
```
[the remark this answers is not in the document]
Comment (doc-han) [reply index.js]: so that the regex can match how I wanted

**Pull request merged**